<a href="https://colab.research.google.com/github/Meyssemmedini/Project-2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

In [ ]:
df= pd.read_csv('/content/drive/MyDrive/coding dojo/Machine Learning/week 7/new_train.csv')
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32950 entries, 0 to 32949
Data columns (total 16 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   age          32950 non-null  int64 
 1   job          32950 non-null  object
 2   marital      32950 non-null  object
 3   education    32950 non-null  object
 4   default      32950 non-null  object
 5   housing      32950 non-null  object
 6   loan         32950 non-null  object
 7   contact      32950 non-null  object
 8   month        32950 non-null  object
 9   day_of_week  32950 non-null  object
 10  duration     32950 non-null  int64 
 11  campaign     32950 non-null  int64 
 12  pdays        32950 non-null  int64 
 13  previous     32950 non-null  int64 
 14  poutcome     32950 non-null  object
 15  y            32950 non-null  object
dtypes: int64(5), object(11)
memory usage: 4.0+ MB


age           job   marital          education  default housing loan  \
0   49   blue-collar   married           basic.9y  unknown      no   no   
1   37  entrepreneur   married  university.degree       no      no   no   
2   78       retired   married           basic.4y       no      no   no   
3   36        admin.   married  university.degree       no     yes   no   
4   59       retired  divorced  university.degree       no      no   no   

     contact month day_of_week  duration  campaign  pdays  previous  \
0   cellular   nov         wed       227         4    999         0   
1  telephone   nov         wed       202         2    999         1   
2   cellular   jul         mon      1148         1    999         0   
3  telephone   may         mon       120         2    999         0   
4   cellular   jun         tue       368         2    999         0   

      poutcome    y  
0  nonexistent   no  
1      failure   no  
2  nonexistent  yes  
3  nonexistent   no  
4  nonexistent   no

#First Choice: Banking Dataset Classification

##Source of Data:

The data comes from direct marketing campaigns of a Portuguese banking institution. [kaggle link](https://www.kaggle.com/datasets/rashmiranu/banking-dataset-classification)

##Brief Description of Data:

The dataset we are using (train.csv) has 32,950 examples and 16 inputs (features).  The target feature is whether the client will subscribe to a term deposit ('yes' or 'no'). The marketing campaigns were based on phone calls, and multiple contacts with the same client were made to determine if they would subscribe to the product.
##Target:

The target variable is 'y,' representing whether the client will subscribe to a term deposit ('yes' or 'no').
##What Does One Row Represent?

One row represents a client who was part of a marketing campaign, and the features capture various details about the client and the campaign.
Classification or Regression Problem:

This is a classification problem since the goal is to predict whether the client will subscribe to a term deposit or not.
##Number of Features:

There are 16 features in the dataset we are using .
##Number of Rows:

The dataset we are usinghas 32,950 rows
##Challenges:

Challenges in cleaning, exploring, or modeling this dataset might include handling categorical features, dealing with any missing or inconsistent data, and understanding the impact of features like 'duration' that highly affect the target variable.

In [ ]:
df2= pd.read_csv("/content/drive/MyDrive/coding dojo/Machine Learning/week 7/investments_VC.csv", encoding= 'latin-1')
df2.info()
df2.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54294 entries, 0 to 54293
Data columns (total 39 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   permalink             49438 non-null  object 
 1   name                  49437 non-null  object 
 2   homepage_url          45989 non-null  object 
 3   category_list         45477 non-null  object 
 4    market               45470 non-null  object 
 5    funding_total_usd    49438 non-null  object 
 6   status                48124 non-null  object 
 7   country_code          44165 non-null  object 
 8   state_code            30161 non-null  object 
 9   region                44165 non-null  object 
 10  city                  43322 non-null  object 
 11  funding_rounds        49438 non-null  float64
 12  founded_at            38554 non-null  object 
 13  founded_month         38482 non-null  object 
 14  founded_quarter       38482 non-null  object 
 15  founded_year       

permalink                name  \
0            /organization/waywire            #waywire   
1  /organization/tv-communications  &TV Communications   
2    /organization/rock-your-paper   'Rock' Your Paper   
3   /organization/in-touch-network   (In)Touch Network   
4   /organization/r-ranch-and-mine  -R- Ranch and Mine   

                    homepage_url  \
0         http://www.waywire.com   
1          http://enjoyandtv.com   
2   http://www.rockyourpaper.org   
3  http://www.InTouchNetwork.com   
4                            NaN   

                                       category_list        market   \
0         |Entertainment|Politics|Social Media|News|          News    
1                                            |Games|         Games    
2                             |Publishing|Education|    Publishing    
3  |Electronics|Guides|Coffee|Restaurants|Music|i...   Electronics    
4                      |Tourism|Entertainment|Games|       Tourism    

   funding_total_usd      status country_code state_code         region  ...  \
0          17,50,000    acquired          USA         NY  New York City  ...   
1          40,00,000   operating          USA         CA    Los Angeles  ...   
2             40,000   operating          EST        NaN        Tallinn  ...   
3          15,00,000   operating          GBR        NaN         London  ...   
4             60,000   operating          USA         TX         Dallas  ...   

  secondary_market  product_crowdfunding round_A round_B round_C  round_D  \
0              0.0                   0.0     0.0     0.0     0.0      0.0   
1              0.0                   0.0     0.0     0.0     0.0      0.0   
2              0.0                   0.0     0.0     0.0     0.0      0.0   
3              0.0                   0.0     0.0     0.0     0.0      0.0   
4              0.0                   0.0     0.0     0.0     0.0      0.0   

  round_E round_F  round_G  round_H  
0     0.0     0.0      0.0      0.0  
1     0.0     0.0      0.0      0.0  
2     0.0     0.0      0.0      0.0  
3     0.0     0.0      0.0      0.0  
4     0.0     0.0      0.0      0.0  

[5 rows x 39 columns]

#Second Choice: Crunchbase Investments Dataset
##Source of Data

The dataset seems to be sourced from Crunchbase, a platform for discovering innovative companies and startups.[kaggle link](https://www.kaggle.com/datasets/arindam235/startup-investments-crunchbase/data)
##Brief Description of Data:

The dataset contains information about investments in companies. has 54293 examples and 39 inputs (features). The challenge is to predict the company status (operating/closed/acquired) based on investment-related criteria and other features.
##Target:

The target variable is the status column, which indicates whether the company is operating, closed, or acquired.
##What Does One Row Represent?

One row represents information about a company, including its name, category, market, funding details, and status and other informations.
##Classification or Regression Problem:

This is a classification problem since the goal is to predict the categorical variable 'status' with multiple classes.
##Number of Features:

The dataset has 39 features
##Number of Rows:
The dataset has 54293 rows
##Challenges:

Challenges in cleaning, exploring, or modeling this dataset might include handling missing values, dealing with categorical variables, understanding the impact of funding details on the company status, and preprocessing text data in columns like category_list and market.

In [ ]:
df3= pd.read_csv('')